### PR TITLE
OOP school library: decorate a class

### PR DESCRIPTION
**Pull request by wandergithub**
Used Decorator design pattern to correct the name of people.
1. Created Interface as class `Nameable` 
3. Prepared base, Capitalize and Trimmer  decorators.

### CODE REVIEW

File: class/person.rb [hunk 1]
```diff
@@ -1,10 +1,14 @@
 require 'securerandom'
+require '../nameable'
+require '../trimmer_decorator'
+require '../capitalize_decorator'
```
Comment (SajjadAhmad14) [inline class/person.rb]: Please use `require_relative` to import local files as `require` is used to import gems and external files.

File: class/person.rb [hunk 2]
```diff
@@ -1,10 +1,14 @@
...
 
-class Person
```
Comment (SajjadAhmad14) [inline class/person.rb:3]: User is unable to correct the name if the characters are more than 10 and also capitalization does not work. Kindly see the screenshot of errors below.

person
es.

trimdecorator
capitaldecorator

**Review by SajjadAhmad14 — CHANGES_REQUESTED** [review]
## Required Changes ♻️
> "Change is inevitable. Change is constant. -Benjamin Disraeli"
## Hi Wander, 
 ### You have been doing a tremendous job so far :ok_hand:

## To Highlight 🎉
Git flow is followed. :heavy_check_mark:
README is professional. :heavy_check_mark:
## To Improve 🙏

- [x] Please check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
#### Please remember to tag me in your question so I can receive the notification.
**_Note: Please do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: class/person.rb (hunk 1 above)
Comment (wandergithub) [reply class/person.rb]: Thanks, @SajjadAhmad14  didn't know that detail 🤝

File: class/person.rb (hunk 2 above)
Comment (wandergithub) [reply class/person.rb:3]: Ohhh yeah, I broke the code solving a linter warning with the `super` call. Solved 👍🏽

**Review by hiromataba — APPROVED** [review]
# Status: Approved :tada: :tada: :tada: 
Hi @wandergithub   :wave: :wave: 

200w  

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**